rootfs-builder/rootfs.sh

```diff
@@ -485,17 +513,41 @@ EOT
 	AGENT_DEST="${AGENT_DIR}/${AGENT_BIN}"
 
 	if [ -z "${AGENT_SOURCE_BIN}" ] ; then
-		info "Pull Agent source code"
-		go get -d "${GO_AGENT_PKG}" || true
-		OK "Pull Agent source code"
-
-		info "Build agent"
-		pushd "${GOPATH_LOCAL}/src/${GO_AGENT_PKG}"
-		[ -n "${AGENT_VERSION}" ] && git checkout "${AGENT_VERSION}" && OK "git checkout successful"
-		make clean
-		make INIT=${AGENT_INIT}
-		make install DESTDIR="${ROOTFS_DIR}" INIT=${AGENT_INIT} SECCOMP=${SECCOMP}
-		popd
+		if [ "$RUST_AGENT" == "no" ]; then
+			info "Pull Agent source code"
+			go get -d "${GO_AGENT_PKG}" || true
+			OK "Pull Agent source code"
+
+			info "Build agent"
+			pushd "${GOPATH_LOCAL}/src/${GO_AGENT_PKG}"
+			[ -n "${AGENT_VERSION}" ] && git checkout "${AGENT_VERSION}" && OK "git checkout successful"
+			make clean
+			make INIT=${AGENT_INIT}
+			make install DESTDIR="${ROOTFS_DIR}" INIT=${AGENT_INIT} SECCOMP=${SECCOMP}
+			popd
+		else
+			# build rust agent
+			info "Build rust agent"
+			# The PATH /.cargo/bin is apparently wrong
+			# looks like $HOME is resolved to empty when
+			# container is started
+			source /root/.cargo/env
+			local -r agent_dir=$(basename ${RUST_AGENT_PKG} .git)
+			pushd "${RUST_SRC_PATH}"
+			if [ ! -d ${RUST_SRC_PATH}/${agent_dir} ]; then
+				git clone https://${RUST_AGENT_PKG}.git
+				cd ${agent_dir}
+				git submodule update --init --recursive
```

Comment:
`cd ${agent_dir}` and `cd ..` are not needed if you use `git -C ${agent_dir}`

Reply:
fixed, thanks! :)